### PR TITLE
Warningscleanup

### PR DESCRIPTION
**Pull request by rbetts**
This branch cleans up Java compiler warnings. Most are harmless. However, the change to HostMessenger looks like a potential crashing defect. The change to ZKUtil fixes a file descriptor leak (in utility code that I don't think is ever called).  Both need more careful review.

### CODE REVIEW

File: src/frontend/org/voltcore/zk/ZKUtil.java [hunk 1]
```diff
@@ -123,14 +123,21 @@ public static File getUploadAsTempFile( ZooKeeper zk, String path, String prefix
 
     public static boolean uploadFileAsChunks(ZooKeeper zk, String zkPath, File file, boolean ephemeral)
             throws Exception {
-        if (file.exists() && file.canRead()) {
-            FileInputStream fis = new FileInputStream(file);
-            ByteBuffer fileBuffer = ByteBuffer.allocate((int)file.length());
-            while (fileBuffer.hasRemaining()) {
-                fis.getChannel().read(fileBuffer);
+        FileInputStream fis = null;
+        try {
```
Comment (BillWhite) [inline src/frontend/org/voltcore/zk/ZKUtil.java]: This looks like a good catch.

**Comment by BillWhite** [comment]
The only question I have is about the removed loggers.  Are they never used?  It looks like they aren't.  I wonder where they came from.  Were they used at one time and then the uses removed?  It probably doesn't matter, but the ability to do logging like this can really help in debugging things ot which may be difficult to attach a debugger.

**Comment by rbetts** [comment]
I didn't remove any log messages - just unused logger variables.

**Comment by BillWhite** [comment]
Well, I was sort of wondering where the messages went.  But perhaps they just slowed things down.

**Comment by BillWhite** [comment]
End of my review comments.  All seems ok to me.

File: src/frontend/org/voltdb/AuthSystem.java [hunk 2]
```diff
@@ -626,12 +626,10 @@ public AuthUser getUser(String name) {
 
         private final String m_user;
         private final byte [] m_password;
-        private ClientAuthHashScheme m_hashScheme;
 
         public HashAuthenticationRequest(final String user, final byte [] hash, final ClientAuthHashScheme scheme) {
```
Comment (nshi) [inline src/frontend/org/voltdb/AuthSystem.java]: You can also remove the `scheme` from the parameters.

File: src/frontend/org/voltdb/ElasticHashinator.java [hunk 3]
```diff
@@ -66,6 +65,9 @@
      */
     private final long m_tokens;
     private final int m_tokenCount;
+
+    // Provide a hook for the GC
```
Comment (nshi) [inline src/frontend/org/voltdb/ElasticHashinator.java]: Thank you, Ryan.

File: src/frontend/org/voltdb/VoltTableRow.java [hunk 4]
```diff
@@ -735,12 +735,15 @@ void putJSONRep(int columnIndex, JSONStringer js) throws JSONException {
...
             break;
+        // VoltType includes a few values that aren't valid column value types
         case INVALID:
             break;
         case NULL:
             break;
         case NUMERIC:
             break;
+        case BOOLEAN:
+            break;
```
Comment (nshi) [inline src/frontend/org/voltdb/VoltTableRow.java]: Interesting. I'm not sure `BOOLEAN` can ever end up in a `VoltTable`. All other switch cases on `VoltType` in `VoltTableRow` don't handle this type.

**Comment by nshi** [comment]
The rest looks good to me. Thanks for cleaning them up, Ryan!

File: src/frontend/org/voltdb/VoltTableRow.java (hunk 4 above)
Comment (paulmartel) [inline src/frontend/org/voltdb/VoltTableRow.java]: fyi, full support for BOOLEAN as a user data type is probably not that far off, so eventually, the other switch cases will have to catch up.

**Pull request merged**